packages/react-combobox/src/components/Combobox/Combobox.types.ts

```diff
@@ -1,19 +1,96 @@
+import * as React from 'react';
+import type { PositioningShorthand } from '@fluentui/react-positioning';
 import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
+import { OptionValue, OptionCollectionState } from '../../utils/OptionCollection.types';
+import { SelectionProps, SelectionState } from '../../utils/Selection.types';
+import { Listbox } from '../Listbox/Listbox';
+import { ComboButton } from '../ComboButton/ComboButton';
 
 export type ComboboxSlots = {
-  root: Slot<'div'>;
+  /* The root combobox slot */
+  root: NonNullable<Slot<'div'>>;
+
+  /* The dropdown listbox slot */
+  listbox: Slot<typeof Listbox>;
+
+  /* The primary slot, the element with role="combobox" */
+  trigger: Slot<typeof ComboButton>;
```

Comment:
⚠️ This is against JSX children pattern that we are trying to follow in other components.

Can we use JSX children instead of these slots to align with other components? For example:

```tsx
<Combobox>
  <ComboButton />
  <Listbox>
   <Option />
   <Option />
  </Listbox>
</Combobox>
```

Reply:
here's an issue for the pattern: https://github.com/microsoft/fluentui/issues/22055